src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+  static deleteIndex = Animal.alive.findIndex((animal) => animal.health <= 0);
+
+  deleteAnimal() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(name);
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.hidden === false && animal instanceof Herbivore) {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
+      animal.deleteAnimal();
+    }
```

Comment:
The `deleteAnimal` method is called on the `animal` instance, but according to the task description, the animal should be removed from the `Animal.alive` array when its health is less than or equal to `0`. This logic should be handled within the `Animal` class, potentially in a method that is called after each time health is modified.